simulator/pkg/debuggablescheduler/debuggable_scheduler.go

```diff
@@ -0,0 +1,184 @@
+package debuggablescheduler
+
+import (
+	"context"
+	"flag"
+	"os"
+
+	"golang.org/x/xerrors"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	componentbaseconfig "k8s.io/component-base/config"
+	_ "k8s.io/component-base/logs/json/register" // for JSON log format registration
+	_ "k8s.io/component-base/metrics/prometheus/clientgo"
+	_ "k8s.io/component-base/metrics/prometheus/version" // for version metric registration
+	"k8s.io/klog/v2"
+	v1 "k8s.io/kube-scheduler/config/v1"
+	"k8s.io/kubernetes/cmd/kube-scheduler/app"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/scheme"
+	configv1 "k8s.io/kubernetes/pkg/scheduler/apis/config/v1"
+	configv1beta2 "k8s.io/kubernetes/pkg/scheduler/apis/config/v1beta2"
+	configv1beta3 "k8s.io/kubernetes/pkg/scheduler/apis/config/v1beta3"
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+
+	simulatorconfig "sigs.k8s.io/kube-scheduler-simulator/simulator/config"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler"
+	simulatorschedulerconfig "sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/config"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/plugin"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/storereflector"
+)
+
+// CreateOptionForOutOfTreePlugin creates the option which can be help with running the external scheduler.
```

Comment:
```suggestion
// CreateOptionForOutOfTreePlugin creates the option which can be help with running the debuggable scheduler.
```
In nearly future, we may be replacing our simulator's internal scheduler with this debuggable scheduler. So this method aims not only external scheduler but also the internal one.